### PR TITLE
Support compiling for Windows with OpenWatcom

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

File: src/load.c
Comment (AliceLR) [inline src/load.c:49]: Going to defer to @sezero for this pull request since I'm not familiar at all with Watcom.

This change is my only (minor) concern and it's more of a question—pt_popen.c should be safe (aside from reentrancy issues that will be addressed soon), whereas the Win32 `_popen` function is apparently questionable at best (source: [MSDN](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/popen-wpopen?view=msvc-160)) (#327). So, my question is, does pt_popen.c work for Watcom/Windows NT (in which case this change shouldn't be necessary)?

If pt_popen.c is fine for Windows NT, adding a `WIN32_OBJS` to the list of things `watcom-prepare` inserts into watcom.mif and appending that to `ALL_OBJS` in Makefile.nt might work.

File: include/xmp.h
Comment (sezero) [inline include/xmp.h:17]: Why don't we define BUILDING_DLL in Makefile.nt, instead?

**Review by sezero — CHANGES_REQUESTED** [review]
Apart from the questions added to particular changes,
`<toplevel>`/watcom.mif.in and lite/watcom.mif.in files are
missing: need adding.

**Comment by sezero** [comment]
> Apart from the questions added to particular changes,
> `<toplevel>`/watcom.mif.in and lite/watcom.mif.in files are
> missing: need adding.

This was my mistake: the file is there, sorry.

File: lite/watcom.mif.in
Comment (sezero) [inline lite/watcom.mif.in:5]: We need `-DBUILDING_DLL` here, too.

Comment (ccawley2011) [reply lite/watcom.mif.in]: OK, fixed.

File: Makefile.nt
Comment (sezero) [inline Makefile.nt:33]: Lose the `src\win32\` path here: The line following it already handles it. Otherwise cross-compile on linux fails.

**Review by sezero — CHANGES_REQUESTED** [review]

**Comment by sezero** [comment]
@AliceLR: The ptpopen issue seems to be resolved. Do you have additional concerns?

**Comment by sezero** [comment]
And what is the codecov failure thing _again_? @cmatsuoka?

**Review by sezero — APPROVED** [review]
This looks OK to me now.

**Comment by sezero** [comment]
Apart from the codecov screwup (@cmatsuoka?), we can merge this if there are no other concerns.

**Review by AliceLR — APPROVED** [review]
This looks OK to me too. Not sure what the codecov issue is but from the log it might just be a missing `sudo apt update`.

**Comment by sezero** [comment]
Then should we force-apply this?

**Comment by AliceLR** [comment]
I think I am OK with merging this as-is if you are.

**Comment by sezero** [comment]
Tried a `re-run` of codecov, still fails:
```
Get:13 http://azure.archive.ubuntu.com/ubuntu focal/main amd64 libgd-perl amd64 2.71-2build1 [130 kB]
Err:11 http://security.ubuntu.com/ubuntu focal-updates/main amd64 python3-lxml amd64 4.5.0-1ubuntu0.2
  404  Not Found [IP: 52.147.219.192 80]
Fetched 685 kB in 0s (2794 kB/s)
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/l/lxml/python3-lxml_4.5.0-1ubuntu0.2_amd64.deb  404  Not Found [IP: 52.147.219.192 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Error: Process completed with exit code 100.
```
Will force-merge this one.

**Pull request merged**
